src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseFVMKernelUtilities.hpp

```diff
@@ -503,17 +549,38 @@ struct C1PPUPhaseFlux
 
     phaseFlux = Ttrans * (potGrad * mobility_i - smoMax * (mobility_j - mobility_i));
 
-    // derivativess
+    if (numDir ==1)
+    {
+      real64 const dMob_dP = dPhaseMob[seri[0]][sesri[0]][sei[0]][ip][Deriv::dP][0];
+      dPhaseFlux_dP[0] += Ttrans * potGrad * dMob_dP;
+    }
+    
+
+    // derivatives
 
     // first part, mobility derivative
 
     // dP
+
+    else
     {
       real64 const dMob_dP = LvArray::tensorOps::AiBi< 3 >( dPhaseMob[seri[0]][sesri[0]][sei[0]][ip][Deriv::dP], faceNormal );
       dPhaseFlux_dP[0] += Ttrans * potGrad * dMob_dP;
     }
 
     // dC
+
+    if (numDir ==1)
```

Comment:
As we are to that many if-branching, I would suggest investing time in refactoring into two functions and one and only `if (numDir==1)`

Reply:
I'll take a look at applying these changes